mkosi/__init__.py

```diff
@@ -2765,7 +2767,22 @@ def install_debian_or_ubuntu(args: MkosiArgs, root: Path, *, do_run_build_script
             if "VERSION_ID" not in os_release and "BUILD_ID" not in os_release:
                 f.write(f"BUILD_ID=mkosi-{args.release}\n")
 
-    invoke_apt(args, do_run_build_script, root, "install", extra_packages)
+    if args.repos_dir:
+        for path in glob.glob(f"{args.repos_dir}/*"):
+            filename = Path(path).name
+            if root.joinpath(f"etc/apt/sources.list.d/{filename}").exists():
+                die(f"Repository file {filename} already exists in image")
+            shutil.copy(path, root / "etc/apt/sources.list.d")
```

Comment:
This won't really work like this, we'd need to mount the repos (or copy their conents) from `args.repos_dir` somewhere in the image, e.g. `/var/lib/mkosi` and add files in `/etc/sources.list.d` of the form 
```
deb file:/var/lib/mkosi/foo {args.release} main contrib non-free
deb-src file:/var/lib/mkosi/foo {args.release} main contrib non-free
```
for some repo `foo/` below `args.repodir`.